SSA/Projects/InstCombine/tests/goals/g2004h02h23hShiftShiftOverflow_proof_test_thm_extracted_1__1.lean

```diff
@@ -0,0 +1,20 @@
+ -- auto-generated from 'SSA/Projects/InstCombine/scripts/extract-goals.py'
+import SSA.Projects.InstCombine.TacticAuto
+import SSA.Projects.InstCombine.LLVM.Semantics
+open BitVec
+open LLVM
+
+set_option linter.unusedTactic false
+set_option linter.unreachableTactic false
+set_option maxHeartbeats 5000000
+set_option maxRecDepth 1000000
+set_option Elab.async false
+
+theorem test_thm.extracted_1._1 : ∀ (x : BitVec 32),
+  ¬(17#32 ≥ ↑32 ∨ 17#32 ≥ ↑32) →
+    31#32 ≥ ↑32 →
+      HRefinement.IsRefinedBy (β := PoisonOr (BitVec 32)) (self :=
+        @instHRefinementOfRefinement _
+          (@PoisonOr.instRefinement _ (@instHRefinementOfRefinement _ InstCombine.instRefinementBitVec)))
+        (PoisonOr.value ((x.sshiftRight' 17#32).sshiftRight' 17#32)) PoisonOr.poison :=
```

Comment:
See that the extraction correctly deals with "ambigious" syntax such as the refinement relation by asking lean to insert typeclass instances wherever appropriate.